openid-4-verifiable-presentations-1_0.md

```diff
@@ -272,7 +272,13 @@ A public key to be used by the Wallet as an input to the key agreement to encryp
 : OPTIONAL. A string determining the HTTP method to be used when the `request_uri` parameter is included in the same request. Two case-sensitive valid values are defined in this specification: `get` and `post`. If `request_uri_method` value is `get`, the Wallet MUST send the request to retrieve the Request Object using the HTTP GET method, i.e., as defined in [@RFC9101]. If `request_uri_method` value is `post`, a supporting Wallet MUST send the request using the HTTP POST method as detailed in (#request_uri_method_post). If the `request_uri_method` parameter is not present, the Wallet MUST process the `request_uri` parameter as defined in [@RFC9101]. Wallets not supporting the `post` method will send a GET request to the request URI (default behavior as defined in [@RFC9101]). `request_uri_method` parameter MUST NOT be present if a `request_uri` parameter is not present.
 
 If the Verifier set the `request_uri_method` parameter value to `post` and there is no other means to convey its capabilities to the Wallet, it SHOULD add the `client_metadata` parameter to the Authorization Request. 
-This enables the Wallet to assess the Verifier's capabilities, allowing it to transmit only the relevant capabilities through the `wallet_metadata` parameter in the Request URI POST request. If the Verifier uses the `client_id_scheme` parameter in the Request Object, it MUST also add the same `client_id_scheme` value in the Authorization Request.   
+This enables the Wallet to assess the Verifier's capabilities, allowing it to transmit only the relevant capabilities through the `wallet_metadata` parameter in the Request URI POST request. If the Verifier uses the `client_id_scheme` parameter in the Request Object, it MUST also add the same `client_id_scheme` value in the Authorization Request. 
+
+`transaction_data`: 
+: OPTIONAL. Array of strings, where each string is a Base64url encoded object that contains a typed parameter set with details about the transaction that the Verifier is requesting the End-User to authorize. See (#transaction_data) for details. The Wallet MUST refuse to process any unknown transaction data type or transaction data not conforming to the respective type definition. Each object consists of the following parameters:
+
+* `type`: REQUIRED. String that is the Identifier of the transaction data type and determines the allowable contents of the object that contains it. The specific values are out of scope of this specification.
+* `input_descriptor_ids`: REQUIRED. Array of strings each pointing to an Input Descriptor that identifies a request for a Credential that the Verifier is requesting transaction data in a particular object to be bound to.
```

Comment:
with the likely coming of the new query language (i.e., #220), do we want to further ingrain the usage of input descriptor at this point?

Reply:
depends on the timeline, I guess? if query language PR gets merged first, we probably need to add another parameter here.

Reply:
yeah, fair point. I was thinking in the other direction though. And knowing how darn hard it is to remove things once they are in, I'd suggest removing this and leaving it implicit for the initial support of the feature.

Reply:
we will need `input_desscriptor_id` as long as presentation exchange is in the specification and we do not have an agreement/timeline to remove presentation exchange, yet until we have that, this kind of parameter is needed for interoperability. and from a priority perspective, transaction data is higher than a new query language right now since there are quite some implementers who are waiting to use transaction data with existing query language.

so 2 options:
1. keep the parameter name `input_descriptor_id` and add a new parameter for a new query language with an equivalent meaning once that query language is in
2. rename this parameter right now to be more general so it can be used with both query languages

option 2 might be better and less changes for the implementations in the future

Reply:
```suggestion
* `requested_credential_ids`: REQUIRED. Array of strings each pointing to an Input Descriptor that identifies a request for a Credential that the Verifier is requesting transaction data in a particular object to be bound to.
```
is this general enough?

Reply:
works for me

Reply:
don't hate it

Reply:
discussed in the WG, since we do not expect both presentation_definition and credentials parameters sent together, so rough consensus to go option 2.